cub/cub/device/device_run_length_encode.cuh

```diff
@@ -224,6 +224,10 @@ struct DeviceRunLengthEncode
 
     using policy_t = detail::device_run_length_encode_policy_hub<accum_t, key_t>;
 
+    if (num_items < 1) {
+      return cudaSuccess;
+    }
```

Comment:
important: I'd expect 0 to be written into `d_num_runs_out`. Why was this change needed? If algorithm can't handle empty problems, let's file an issue and address it in a separate PR.

Reply:
Yeah that is something I was wondering about too.

Funnily enough NonTrivialRuns does not write anything which definitely is a bug

Reply:
Thank you for verifying that! Let's file an issue and disable the test case for now.

Reply:
If I understand you correctly, I will remove the changes to `device_run_length_encode.cuh` and ifdef the tests out.

We will then device a proper fix in a follow up (same as the 64bit issue with NonTrivialRuns)

Reply:
Below we pass in `lengths_input_iterator_t((length_t)1),` which requires at least a single element.

I was wondering about changing that to `lengths_input_iterator_t((length_t)(min)(1, num_items)),` But I would have to verify that it does not break ReduceByKey in an interesting way

Reply:
I adopted the functions to properly write `d_num_runs_out`